README.md

```diff
@@ -1,3 +1,142 @@
+
+# Config.load_prefixed_env_vars(allowed_prefixes: List[str])
+
+Load environment variables with specified prefixes and store them in the `Config` object.
+
+- **Args**:
+  - `allowed_prefixes (list)`: A list of allowed prefixes for environment variables.
+- **Raises**:
+  - `TypeError`: If allowed_prefixes is not provided.
+
+# Config.load_from_yaml_file(filepath: str)
+
+Load configuration data from a YAML file and store it in the `Config` object.
+The YAML can have values prefixed with `VAULT.` and `ENV.`.
+
+```yaml
+url: "stage"
+database: "db_address"
+db_pass: "VAULT.vault_secret_path.vault_secret_key"
+username: "ENV.USER"
+
+cloudaccessdb:
+  prefix_name: "cloud_db"
+  user: "cloud_access_user"
+  host: "example.com"
+
+cloud_access_db:
+  port: 1111
+  dbName: "cloud_access"
+```
+
+- **Args**:
+  - `filepath (str)`: The path to the YAML file containing the configuration data.
+- **Raises**:
+  - `FileNotFoundError`: If the specified file is not found.
+  - `ValueError`: If the file content is not valid YAML.
+  - `yaml.YAMLError`: If there is an error parsing the YAML data.
+
+## How to use the VAULT prefix within YAML
+
+The value prefixed with `VAULT` must be dot separated and followed respectively by Vault's path and Vault's secret key.
+The key saved in the `Config` object is the one pointed out as key in the YAML file. In the YAML example above it's `db_pass`.
+
+To use this method, `Config` object must be initialized with a `VaultFetcher` object that satisfies the Interface example:
+
+```python
+from abc import ABC, abstractmethod
+
+class VaultFetcherInterface(ABC):
+    @abstractmethod
+    def get_value_from_vault(self, path: str, key: str) -> str:
+        """
+        Abstract method to fetch a secret value from Vault.
+
+        Args:
+            path (str): The path to the secret in Vault.
+            key (str): The key of the secret within the specified path.
+
+        Returns:
+            str: The secret value fetched from Vault.
+        """
+        pass
+```
+
+## How to use the ENV prefix within YAML
+
+The value prefixed with `ENV` must be dot separated and followed by the environment variable key that should be loaded.
+
+# Config.load_from_vault(vault_secret_path: str, vault_secret_key: str, fetcher: Any, custom_key_name: str = None)
+Load a secret value from Vault and store it in the Config object.
+
+To use this method, `Config` object must also be initialized with a `VaultFetcher` (see example above).
+
+- **Args**:
+    - `vault_secret_path (str)`: The path to the secret in Vault.
+    - `vault_secret_key (str)`: The key of the secret within the specified path.
+    - `custom_key_name (str, optional)`: The custom key name to use in the Config object. If not provided, the vault_secret_key is used as the key name.
+- **Raises**:
+    - `KeyError`: If the specified Vault secret path or key is not found.
+
+# Config.__setitem__(key: str, value: str)
```

Comment:
@andressadotpy , the `__` syntax is conflicting with make file preview and this is appearing as `Config.setitem` in the preview